### PR TITLE
For #11800: Hide reveal and clear password icons if the password is empty when editing a saved login

### PR DESCRIPTION
**Pull request by joc-a**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

Before fix - Light Mode
Before fix - Light Mode

Before fix - Dark Mode
Before fix - Dark Mode

After fix - Light Mode
After fix - Light Mode

After fix - Dark Mode
After fix - Dark Mode

### CODE REVIEW

**Comment by ekager** [comment]
Hi @joc-a thanks for picking this up!

So we also set errors on the username state if you have 2 logins for a site and edit one login to match the other username (ie username dupe). Let's also verify that looks correct. 

Secondly, I think we also want to get the error icon vertically aligned with the clear icon/reveal icon to match the mocks. We might be able to do this through some drawable padding (either in the layout or we can create a new layer-list drawable to pad the error icon as we need).

Let me know if this makes sense and if you have any other questions!

**Comment by joc-a** [comment]
Hi @ekager !

I aligned the error icon for the password. It now looks like this:
After fix - Light Mode
After fix - Dark Mode


I will do the same thing for the username but first I'd like to know what we want it to look like when the username is a duplicate. It currently looks like this:
Screenshot_1600721223


Also, do we want to allow saving an empty username?
Screenshot_1600721227

**Comment by ekager** [comment]
Thanks @joc-a this is looking great!
That's a very good question! 🤔 cc @apbitner from our UX team and @eliserichards as the eng who originally worked on this to answer these 2 questions:
1. What should the error state of the username field look like with the error icon and clear icon displayed?
2. Do we want to allow empty usernames to be saved? @eliserichards do you remember?

**Comment by apbitner** [comment]
> What should the error state of the username field look like with the error icon and clear icon displayed?
It seems to me that for the password error state we are not showing the clean icon and only showing the error icon, is that correct? If so, we might as well use the same pattern for username.

**Comment by eliserichards** [comment]
> Thanks @joc-a this is looking great!
> That's a very good question! 🤔 cc @apbitner from our UX team and @eliserichards as the eng who originally worked on this to answer these 2 questions:
> 
>     1. What should the error state of the username field look like with the error icon and clear icon displayed?

The only error state for username is if the username entered would make the login a duplicate of another login that the user has saved. 

>     2. Do we want to allow empty usernames to be saved? @eliserichards do you remember?

Empty usernames are valid (as an empty string, it should never be`null`)

**Comment by eliserichards** [comment]
I agree with @apbitner that we shouldn't show the username `clear` icon if there's a username error 👍

File: app/src/main/res/drawable/mozac_ic_warning_with_bottom_padding.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (ekager) [inline app/src/main/res/drawable/mozac_ic_warning_with_bottom_padding.xml:1]: nit - this needs license text at the top, look at any other xml file and copy it :)

File: app/src/main/java/org/mozilla/fenix/settings/logins/fragment/EditLoginFragment.kt [hunk 2]
```diff
@@ -231,21 +239,26 @@ class EditLoginFragment : Fragment(R.layout.fragment_edit_login) {
...
         }
     }
 
     private fun setPasswordError() {
         inputLayoutPassword?.let { layout ->
             validPassword = false
             layout.error = context?.getString(R.string.saved_login_password_required)
-            layout.setErrorIconDrawable(R.drawable.mozac_ic_warning)
+            layout.setErrorIconDrawable(R.drawable.mozac_ic_warning_with_bottom_padding)
+            layout.setErrorIconTintList(ColorStateList.valueOf(ContextCompat.getColor(requireContext(), R.color.design_error)))
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/settings/logins/fragment/EditLoginFragment.kt:261]: L244 and L261 are hitting the max line length error, you can autoformat with Android Studio or just manually break it into multiple lines!

**Review by ekager — CHANGES_REQUESTED** [review]
Looking good! Just pulled it down and left 2 comments of things we need to change.

**Review by ekager — APPROVED** [review]
Looks good! Thanks for the contribution!

**Comment by joc-a** [comment]
This was my first time contributing to Fenix and my second time contributing on GitHub. Thanks for being friendly and making it a pleasant experience! @ekager 
I do have a question: can external contributors rebase on master when they're done, or is that something you'd rather do internally?

**Comment by ekager** [comment]
I can squash and merge when the PR passes CI!

**Pull request merged**